source/seedsource/management/commands/import_seed_zones.py

```diff
@@ -115,77 +92,60 @@ def _add_zones(self, source, name, uid, species, path, zone_field):
 
                         uid_suffix += 1
 
+    def _get_historic_name(self, zone_id, object_id):
+        # Special case: there are two zone 842s: one in WA and the other in OR
+        if zone_id == 842:
+            if object_id == 28:
+                state = 'Washington'
+            else:
+                state = 'Oregon'
+
+        elif zone_id in WA_HISTORIC:
+            state = 'Washington'
+        elif zone_id in OR_HISTORIC:
+            state = 'Oregon'
+        elif zone_id in WA_OR_HISTORIC:
+            state = 'Oregon & Washington'
+        else:
+            raise ValueError('Could not find state for zone {}'.format(zone_id))
+
+        return '{} (1966/1973) Zone {:03d}'.format(state, zone_id)
+
     def handle(self, zones_file, *args, **options):
         temp_dir = mkdtemp()
 
+        zones_file = zones_file[0]
+
         try:
-            with ZipFile(zones_file[0]) as zf:
+            with ZipFile(zones_file) as zf:
                 zf.extractall(temp_dir)
 
-            wa_zones = os.path.join(temp_dir, WASHINGTON_ZONES_DIR)
-            or_zones = os.path.join(temp_dir, OREGON_ZONES_DIR)
-            historic_zones = os.path.join(temp_dir, HISTORIC_ZONES_DIR)
-            california_zones = os.path.join(temp_dir, CALIFORNIA_ZONES_DIR)
+            with open(os.path.join(temp_dir, 'config.json'), 'r') as f:
+                config = json.loads(f.read())
 
             message = (
-                'WARNING: This will replace all your seed zone records and remove associated transfer limits. '
-                'Do you want to continue? [y/n]'
+                'WARNING: This will replace {} seed zone records and remove associated transfer limits. '
+                'Do you want to continue? [y/n]'.format(config['label'])
             )
             if input(message).lower() not in {'y', 'yes'}:
                 return
 
             with transaction.atomic():
-                SeedZone.objects.all().delete()
+                SeedZone.objects.filter(source__startswith=config['dir']).delete()
 
-                print('Loading Washington seed zones...')
+                print('Loading {} seed zones...'.format(config['label']))
 
-                for species, (zone_file, name) in WASHINGTON_ZONES.items():
-                    self._add_zones(
-                        os.path.join(WASHINGTON_ZONES_DIR, zone_file), name, 'wa_{}_{{zone_id}}'.format(species),
-                        species, os.path.join(wa_zones, zone_file), 'ZONE_NO'
-                    )
+                zones = os.path.join(temp_dir, config['dir'])
 
-                print('Loading Oregon seed zones...')
+                for species, (zone_file, name, uid, col_name, bands_fn) in config['species'].items():
```

Comment:
Please use key/value pairs for the items inside the `species` object in the JSON so that it's a little more self-documenting. E.g.,

```
"psme": [
  "file": "PSME.shp",
  "label": "Washington (2002) Douglas-fir Zone {zone_id}",
  "name": "wa_psme_{zone_id}",
  "column": "ZONE_NO",
  "bands_fn": "wa_psme"
],
```